src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -269,21 +283,39 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
-                $parameters,
-                $requestBody,
-                isset($operation->getOpenapiContext()['callbacks']) ? new \ArrayObject($operation->getOpenapiContext()['callbacks']) : null,
-                $operation->getOpenapiContext()['deprecated'] ?? (bool) $operation->getDeprecationReason(),
-                $operation->getOpenapiContext()['security'] ?? null,
-                $operation->getOpenapiContext()['servers'] ?? null,
-                array_filter($operation->getOpenapiContext() ?? [], static fn ($item): int|false => preg_match('/^x-.*$/i', (string) $item), \ARRAY_FILTER_USE_KEY)
-            ));
+            $openapiOperation = ($operation->getOpenapi() ?: new Model\Operation())
+                // Defaults
+                ->withTags($operation->getOpenapi()?->getTags() ?: [$operation->getShortName() ?: $resourceShortName])
+                ->withSummary($operation->getOpenapi()?->getSummary() ?: $this->getPathDescription($resourceShortName, $method, $operation instanceof CollectionOperationInterface))
+                ->withDescription($operation->getOpenapi()?->getDescription() ?: $this->getPathDescription($resourceShortName, $method, $operation instanceof CollectionOperationInterface))
+                ->withDeprecated($operation->getOpenapi()?->getDeprecated() ?: (bool) $operation->getDeprecationReason());
+
+            // TODO Remove in 4.0
+            if (null !== $operation->getOpenapiContext()) {
+                $allowedProperties = array_map(fn (\ReflectionProperty $reflProperty): string => $reflProperty->getName(), (new \ReflectionClass(Model\Operation::class))->getProperties());
+                foreach ($operation->getOpenapiContext() as $key => $value) {
+                    $value = match ($key) {
+                        'externalDocs' => new ExternalDocumentation(description: $value['description'] ?? '', url: $value['url'] ?? ''),
+                        'requestBody' => new RequestBody(description: $value['description'] ?? '', content: isset($value['content']) ? new \ArrayObject($value['content'] ?? []) : null, required: $value['required'] ?? false),
+                        'callbacks' => new \ArrayObject($value ?? []),
+                        default => $value,
+                    };
+
+                    if (\in_array($key, $allowedProperties, true)) {
+                        $openapiOperation = $openapiOperation->{'with'.ucfirst($key)}($value);
+                        continue;
+                    }
+
+                    $openapiOperation = $openapiOperation->withExtensionProperty((string) $key, $value);
+                }
+            }
+
+            $pathItem = $pathItem->{'with'.ucfirst($method)}($openapiOperation
+                ->withOperationId($operationId)
+                ->withResponses($responses)
+                ->withParameters($parameters)
+                ->withRequestBody($requestBody)
+            );
 
             $paths->addPath($path, $pathItem);
         }
```

Comment:
We should add a deprecation here don't you think? If the openapicontext array has some properties, just `trigger_deprecation`.